bot/exts/christmas/adventofcode.py

```diff
@@ -301,29 +521,60 @@ async def private_leaderboard_daily_stats(self, ctx: commands.Context) -> None:
         Embed will display the total members and the number of users who have completed each day's puzzle
         """
         async with ctx.typing():
-            await self._check_leaderboard_cache(ctx)
-
-            if not self.cached_private_leaderboard:
-                # Feedback on issues with leaderboard caching are sent by _check_leaderboard_cache()
-                # Short circuit here if there's an issue
-                return
+            is_staff = ctx.channel.id == Channels.advent_of_code_staff
+            if is_staff:
+                await self.check_staff_leaderboard()
+                if self.cached_staff_leaderboard is None:
+                    await ctx.send(":x: Missing leaderboard cache.")
+                    return
+            else:
+                await self.check_leaderboard()
 
             # Build ASCII table
-            total_members = len(self.cached_private_leaderboard.members)
+            if is_staff:
+                total_members = len(self.cached_staff_leaderboard.members)
+            else:
+                total_members = await self.public_user_data.length()
+                if total_members == 0:
+                    await ctx.send(":x: Missing leaderboard cache. Please notify staff about it.")
+                    return
+
             _star = Emojis.star
             header = f"{'Day':4}{_star:^8}{_star*2:^4}{'% ' + _star:^8}{'% ' + _star*2:^4}\n{'='*35}"
             table = ""
-            for day, completions in enumerate(self.cached_private_leaderboard.daily_completion_summary):
-                per_one_star = f"{(completions[0]/total_members)*100:.2f}"
-                per_two_star = f"{(completions[1]/total_members)*100:.2f}"
+            if is_staff:
+                for day, completions in enumerate(self.cached_staff_leaderboard.daily_completion_summary):
+                    per_one_star = f"{(completions[0]/total_members)*100:.2f}"
+                    per_two_star = f"{(completions[1]/total_members)*100:.2f}"
 
-                table += f"{day+1:3}){completions[0]:^8}{completions[1]:^6}{per_one_star:^10}{per_two_star:^6}\n"
+                    table += f"{day+1:3}){completions[0]:^8}{completions[1]:^6}{per_one_star:^10}{per_two_star:^6}\n"
```

Comment:
Same as my earlier comment, let `enumerate` do the work for you
```suggestion
                for day, completions in enumerate(self.cached_staff_leaderboard.daily_completion_summary, 1):
                    per_one_star = f"{(completions[0]/total_members)*100:.2f}"
                    per_two_star = f"{(completions[1]/total_members)*100:.2f}"

                    table += f"{day:3}){completions[0]:^8}{completions[1]:^6}{per_one_star:^10}{per_two_star:^6}\n"
```